Possible issue with re-orgs?

* **I'm submitting a ...**
    - [ ] feature request
    - [x] bug report

* **Bug Report**

Our Ropsten node has failed with the following exception:

```
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:560)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:321)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:224)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:192)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:175)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:219)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.kafka.connect.errors.ConnectException: java.sql.SQLException: java.sql.BatchUpdateException: Batch entry 133 INSERT INTO "transaction" ("hash","nonce","block_hash","block_number","transaction_index","from","to","value","gas_price","gas","input","v","r","s","timestamp","creates","chain_id") VALUES ('0x40c4646b95175d896b64442d77192b0429afab38877f08f7632bbd71f4ada644','5384','0x3b8dea936081bd68be2bf2fd89962d8fe4988b90aab6b3d83bb94c3415566d10','5681316',0,'0x0002851146112cef5d360033758c470689b72ea7','0xdc3422c75a04e64c30b4cedac699239d48bfba35','0','80000000000','6308598',?,41,'0x8c5c47cd85dc19f483e1e16175d21d1bcd5c168523684e51941b034d01195262','0x770edd87b86d7c1b28c92158ab0df5fd2447db0b7643947b33b22a61843b7c2f',1558960723,NULL,3) ON CONFLICT ("hash") DO UPDATE SET "nonce"=EXCLUDED."nonce","block_hash"=EXCLUDED."block_hash","block_number"=EXCLUDED."block_number","transaction_index"=EXCLUDED."transaction_index","from"=EXCLUDED."from","to"=EXCLUDED."to","value"=EXCLUDED."value","gas_price"=EXCLUDED."gas_price","gas"=EXCLUDED."gas","input"=EXCLUDED."input","v"=EXCLUDED."v","r"=EXCLUDED."r","s"=EXCLUDED."s","timestamp"=EXCLUDED."timestamp","creates"=EXCLUDED."creates","chain_id"=EXCLUDED."chain_id" was aborted: ERROR: duplicate key value violates unique constraint "transaction_from_nonce_key"
  Detail: Key ("from", nonce)=(0x0002851146112cef5d360033758c470689b72ea7, 5384) already exists.  Call getNextException to see other errors in the batch.
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "transaction_from_nonce_key"
  Detail: Key ("from", nonce)=(0x0002851146112cef5d360033758c470689b72ea7, 5384) already exists.
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "transaction_from_nonce_key"
  Detail: Key ("from", nonce)=(0x0002851146112cef5d360033758c470689b72ea7, 5384) already exists.

	at io.confluent.connect.jdbc.sink.JdbcSinkTask.put(JdbcSinkTask.java:87)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:538)
	... 10 more
Caused by: java.sql.SQLException: java.sql.BatchUpdateException: Batch entry 133 INSERT INTO "transaction" ("hash","nonce","block_hash","block_number","transaction_index","from","to","value","gas_price","gas","input","v","r","s","timestamp","creates","chain_id") VALUES ('0x40c4646b95175d896b64442d77192b0429afab38877f08f7632bbd71f4ada644','5384','0x3b8dea936081bd68be2bf2fd89962d8fe4988b90aab6b3d83bb94c3415566d10','5681316',0,'0x0002851146112cef5d360033758c470689b72ea7','0xdc3422c75a04e64c30b4cedac699239d48bfba35','0','80000000000','6308598',?,41,'0x8c5c47cd85dc19f483e1e16175d21d1bcd5c168523684e51941b034d01195262','0x770edd87b86d7c1b28c92158ab0df5fd2447db0b7643947b33b22a61843b7c2f',1558960723,NULL,3) ON CONFLICT ("hash") DO UPDATE SET "nonce"=EXCLUDED."nonce","block_hash"=EXCLUDED."block_hash","block_number"=EXCLUDED."block_number","transaction_index"=EXCLUDED."transaction_index","from"=EXCLUDED."from","to"=EXCLUDED."to","value"=EXCLUDED."value","gas_price"=EXCLUDED."gas_price","gas"=EXCLUDED."gas","input"=EXCLUDED."input","v"=EXCLUDED."v","r"=EXCLUDED."r","s"=EXCLUDED."s","timestamp"=EXCLUDED."timestamp","creates"=EXCLUDED."creates","chain_id"=EXCLUDED."chain_id" was aborted: ERROR: duplicate key value violates unique constraint "transaction_from_nonce_key"
  Detail: Key ("from", nonce)=(0x0002851146112cef5d360033758c470689b72ea7, 5384) already exists.  Call getNextException to see other errors in the batch.
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "transaction_from_nonce_key"
  Detail: Key ("from", nonce)=(0x0002851146112cef5d360033758c470689b72ea7, 5384) already exists.
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "transaction_from_nonce_key"
  Detail: Key ("from", nonce)=(0x0002851146112cef5d360033758c470689b72ea7, 5384) already exists.
```
I added as well an screenshot:

![2019-05-27-210015_3439x1230_scrot](https://user-images.githubusercontent.com/82811/58435777-80894000-80c2-11e9-9fc7-13e8441bcd1c.png)

--- a/apps/migrator/sql/V1__Create_basic_tables.sql
+++ b/apps/migrator/sql/V1__Create_basic_tables.sql
@@ -124,8 +124,7 @@ CREATE TABLE "transaction"
   s                 CHAR(78) NOT NULL,
   timestamp         BIGINT   NOT NULL,
   creates           CHAR(42) NULL,
-  chain_id          BIGINT   NULL,
-  UNIQUE ("from", nonce)
+  chain_id          BIGINT   NULL
 );
 
 CREATE INDEX idx_transaction_hash ON TRANSACTION (hash);
@@ -241,10 +240,10 @@ WHERE cb.number IS NOT NULL
 /* All traces including possible traces from old forks */
 CREATE TABLE transaction_trace
 (
-  block_hash           CHAR(66)     NOT NULL,
-  transaction_hash     CHAR(66)     NULL,
-  root_error           VARCHAR(514) NULL,
-  traces               TEXT         NOT NULL,
+  block_hash       CHAR(66)     NOT NULL,
+  transaction_hash CHAR(66)     NULL,
+  root_error       VARCHAR(514) NULL,
+  traces           TEXT         NOT NULL,
   UNIQUE (block_hash, transaction_hash)
 );
 
@@ -460,12 +459,12 @@ SELECT fb.address,
        (SELECT COUNT(*)
         FROM canonical_transaction AS ct
         WHERE ct.from = fb.address) AS out_tx_count,
-        CASE
-          WHEN cont.creator IS NULL THEN
-            FALSE
-          ELSE
-            TRUE
-          END AS is_contract,
+       CASE
+         WHEN cont.creator IS NULL THEN
+           FALSE
+         ELSE
+           TRUE
+         END                        AS is_contract,
        CASE
          WHEN a.count > 0 THEN
            TRUE
@@ -618,7 +617,7 @@ CREATE TABLE token_exchange_rates
 CREATE VIEW canonical_token_exchange_rates AS
 SELECT ter.*
 FROM token_exchange_rates AS ter
-    INNER JOIN canonical_contract AS cc on ter.address = cc.address;
+       INNER JOIN canonical_contract AS cc on ter.address = cc.address;
 
 /* Coin exchange rates table */
 CREATE TABLE coin_exchange_rates

